pkg/types/types.go

```diff
@@ -0,0 +1,368 @@
+	} `json:"containers"`
+}
+
+func GetEnv(envStr string) (Env, error) {
+	switch envStr {
+	case "baremetal":
+		return Baremetal, nil
+	case "cloud":
+		return Cloud, nil
+	default:
+		return -1, fmt.Errorf("invalid cluster environment: %s", envStr)
+	}
+}
+
+func GetDeployment(deploymentStr string) (Deployment, error) {
+	switch deploymentStr {
+	case "mno":
+		return MNO, nil
+	case "sno":
+		return SNO, nil
+	default:
+		return -1, fmt.Errorf("invalid deployment type: %s", deploymentStr)
+	}
+}
+
+func (m *ComMatrix) ToCSV() ([]byte, error) {
+	out := make([]byte, 0)
+	w := bytes.NewBuffer(out)
+	csvwriter := csv.NewWriter(w)
+
+	err := gocsv.MarshalCSV(&m.Matrix, csvwriter)
+	if err != nil {
+		return nil, err
+	}
+
+	csvwriter.Flush()
+
+	return w.Bytes(), nil
+}
+
+func (m *ComMatrix) ToJSON() ([]byte, error) {
+	out, err := json.MarshalIndent(m.Matrix, "", "    ")
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func (m *ComMatrix) ToYAML() ([]byte, error) {
+	out, err := yaml.Marshal(m)
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func (m *ComMatrix) String() string {
+	var result strings.Builder
+	for _, details := range m.Matrix {
+		result.WriteString(details.String() + "\n")
+	}
+
+	return result.String()
+}
+
+func (m *ComMatrix) WriteMatrixToFileByType(utilsHelpers utils.UtilsInterface, fileNamePrefix, format string, deployment Deployment, destDir string) error {
+	if format == FormatNFT {
+		masterMatrix, workerMatrix := m.separateMatrixByRole()
+		err := masterMatrix.writeMatrixToFile(utilsHelpers, fileNamePrefix+"-master", format, destDir)
+		if err != nil {
+			return err
+		}
+		if deployment == MNO {
+			err := workerMatrix.writeMatrixToFile(utilsHelpers, fileNamePrefix+"-worker", format, destDir)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	err := m.writeMatrixToFile(utilsHelpers, fileNamePrefix, format, destDir)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GenerateMatrixDiff generates the diff between mat1 to mat2.
+func (m *ComMatrix) GenerateMatrixDiff(mat2 *ComMatrix) (string, error) {
+	colNames, err := getComMatrixHeadersByFormat(FormatCSV)
+	if err != nil {
+		return "", fmt.Errorf("error getting commatrix CSV tags: %v", err)
+	}
+
+	diff := colNames + "\n"
+
+	for _, cd := range m.Matrix {
+		if mat2.Contains(cd) {
+			diff += fmt.Sprintf("%s\n", cd)
+			continue
+		}
+
+		// add "+" before cd's mat1 contains but mat2 doesn't
+		diff += fmt.Sprintf("+ %s\n", cd)
+	}
+
+	for _, cd := range mat2.Matrix {
+		// Skip "rpc.statd" ports, these are randomly open ports on the node,
+		// no need to mention them in the matrix diff
+		if cd.Service == "rpc.statd" {
+			continue
+		}
+
+		if !m.Contains(cd) {
+			// add "-" before cd's mat1 doesn't contain but mat2 does
+			diff += fmt.Sprintf("- %s\n", cd)
+		}
+	}
+
+	return diff, nil
+}
+
+func (m *ComMatrix) print(format string) ([]byte, error) {
+	switch format {
+	case "json":
```

Comment:
u can here use the 

	FormatJSON = "json"
	FormatYAML = "yaml"
	FormatCSV  = "csv"
	FormatNFT  = "nft"

Reply:
right!